### PR TITLE
Working starting point for FL

### PR DESCRIPTION
**Pull request by viktorvaladi**
Rough starting point that needs some more work for FL. The attack is working here and moved to example/ with some refactoring / additions on functionality

### CODE REVIEW

File: examples/gia/cifar100_inverting_16_images/model.py
Comment (mattiasakesson) [inline examples/gia/cifar100_inverting_16_images/model.py:1]: can't we have models in a subdirectory under examples to get rid of duplications?

File: examples/gia/cifar10_inverting_1_image/cifar.py
Comment (mattiasakesson) [inline examples/gia/cifar10_inverting_1_image/cifar.py:1]: can't we have dataset in a subdirectory under examples to get rid of duplications for every example?

File: leakpro/attacks/gia_attacks/invertinggradients.py
Comment (mattiasakesson) [inline leakpro/attacks/gia_attacks/invertinggradients.py:1]: maybe add the individual settings in init for class InvertingGradients?

**Review by mattiasakesson — CHANGES_REQUESTED** [review]
I added some design questions, but overall everything looks good.

File: leakpro/attacks/gia_attacks/invertinggradients.py
Comment (viktorvaladi) [reply leakpro/attacks/gia_attacks/invertinggradients.py:1]: Could you elaborate?

File: examples/gia/cifar100_inverting_16_images/model.py
Comment (viktorvaladi) [reply examples/gia/cifar100_inverting_16_images/model.py:1]: A good point, but I would argue that for transparency it's better to have each example completely standalone with as few dependencies to our framework as possible. So each example folder should clearly present the parts that we expect a user to supply to run LeakPro. I think in this case repetition of code is OK

File: examples/gia/cifar10_inverting_1_image/cifar.py
Comment (viktorvaladi) [reply examples/gia/cifar10_inverting_1_image/cifar.py:1]: Same as above (and also the cifars vary a bit in their implementation)

File: leakpro/attacks/gia_attacks/invertinggradients.py
Comment (mattiasakesson) [reply leakpro/attacks/gia_attacks/invertinggradients.py:1]: like 
InvertingConfig(total_variation = 1.0e-06, attack_lr = 0.1, at_iterations = 8000, optimizer = MetaSGD(),criterion = CrossEntropyLoss(), epochs = 1):
   self.total_variation=total_variation
   ...
then you could have the possibility to build the object in one line:
InvertingConfig(kvarg)

Comment (viktorvaladi) [reply leakpro/attacks/gia_attacks/invertinggradients.py:1]: Good point. I pushed adding types to the config. Then the @dataclass decorator handles the rest, so now we can initialize it like this:
`config = InvertingConfig(total_variation = 1.0e-04)`
:)

**Review by mattiasakesson — APPROVED** [review]

**Pull request merged**